src/Raven.Server/Documents/Subscriptions/SubscriptionConnectionState.cs

```diff
@@ -123,7 +123,14 @@ public void EndConnection()
 
         public void Dispose()
         {
-            EndConnection();
+            try
+            {
+                EndConnection();
```

Comment:
Need to verify that even on error, everything is properly disposed, especially the unmanaged stuff